cmd/antrea-controller/controller.go

```diff
@@ -94,22 +99,62 @@ func run(o *Options) error {
 
 	go apiServer.GenericAPIServer.PrepareRun().Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go initializePrometheusMetrics(
+			o.config.EnablePrometheusGoMetrics,
+			o.config.EnablePrometheusGoMetrics)
+	}
+
 	<-stopCh
 	klog.Info("Stopping Antrea controller")
 	return nil
 }
 
+// Initialize Prometheus listener and metrics collection.
+func initializePrometheusMetrics(
+	enablePrometheusGoMetrics bool,
+	enablePrometheusProcessMetrics bool) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		klog.Errorf("Failed to retrieve agent node name, %v", err)
+	}
+
+	klog.Info("Initializing prometheus")
+	gaugeHost := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "antrea_controller_host",
+		Help: "Antrea controller hostname (as a label), typically used in grouping/aggregating stats; " +
+			"the label defaults to the hostname of the host but can be overridden by configuration. " +
+			"The value of the gauge is always set to 1.",
+		ConstLabels: prometheus.Labels{"host": hostname},
+	})
+	gaugeHost.Set(1)
+	prometheus.MustRegister(gaugeHost)
+	http.Handle("/metrics", promhttp.Handler())
+
+	if !enablePrometheusGoMetrics {
+		klog.Info("Golang metrics are disabled")
+		prometheus.Unregister(prometheus.NewGoCollector())
+	}
+	if !enablePrometheusProcessMetrics {
+		klog.Info("Process metrics are disabled")
+		prometheus.Unregister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	}
+}
+
 func createAPIServerConfig(kubeconfig string,
 	addressGroupStore storage.Interface,
 	appliedToGroupStore storage.Interface,
-	networkPolicyStore storage.Interface) (*apiserver.Config, error) {
+	networkPolicyStore storage.Interface,
+	enablePrometheusMetrics bool) (*apiserver.Config, error) {
 	// TODO:
 	// 1. Support user-provided certificate.
 	// 2. Support configurable https port.
 	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
 	authentication := genericoptions.NewDelegatingAuthenticationOptions()
 	authorization := genericoptions.NewDelegatingAuthorizationOptions()
-
+	if enablePrometheusMetrics {
+		authorization.WithAlwaysAllowPaths("/metrics")
```

Comment:
Did not figure out from the code how apiserver and prometheus http handler are associated, and the handler will handle "/metrics". Could you explain a little for me?